### ISSUE TITLE
Elgg/Application/CacheHandler->setupSimplecache can override values of settings.php

### ISSUE THREAD

**Issue by jeabakker** [Elgg/Elgg#10351]
https://github.com/Elgg/Elgg/blob/2.x/engine/classes/Elgg/Application/CacheHandler.php#L211

If i have configured 'dataroot' and 'simplecache_enabled' in settings.php but NOT 'cacheroot' the settings get overridden by the database settings.

This can cause problems if the database settings aren't the same as defined in settings.php

https://github.com/Elgg/Elgg/blob/2.x/engine/classes/Elgg/Application/CacheHandler.php#L242
should probably check 

``` php
if (!$config->getVolatile($row->name)) {
  $config->set($row->name, $row->value);
}
```

this would prevent https://github.com/Elgg/Elgg/blob/2.x/engine/classes/Elgg/Application/CacheHandler.php#L245 from being set to a different dir as defined in settings.php

**Comment by mrclay** [comment]
Imagine the code and headaches we could remove by forcing these values to be set in settings.php.

**Comment by hypeJunction** [comment]
> Imagine the code and headaches we could remove by forcing these values to be set in settings.php.

And drop the installer :)

**Comment by mrclay** [comment]
Related #10355

**Comment by jdalsem** [comment]
@mrclay am i correct that this issue can be closed? because of #10894

**Comment by mrclay** [comment]
Yep!
